routing/router_test.go

```diff
@@ -1964,9 +1979,10 @@ func TestDisconnectedBlocks(t *testing.T) {
 		ctx.chain.setBestBlock(int32(height))
 		ctx.chainView.notifyBlock(block.BlockHash(), height,
 			block.Transactions)
+
+		time.Sleep(time.Millisecond * 500)
```

Comment:
Rather than add these sleeps, you could update the mock notifier to ACK once it's sent this block (or quit)

eg, update mock chain to have a `notifyBlockAck` channel:
```
type mockChain struct {
    ...
    notifyBlockAck chan struct{}
}
```

In tests where we want the test to wait until we've consumed the block, set a non-nil channel:
`ctx.chainView.notifyBlockAck= make(chan struct{},1)`

Then the relevant call can send an ack if this channel is non-nil:
```
func (m *mockChainView) notifyBlock(hash chainhash.Hash, height uint32,
	txns []*wire.MsgTx) {

	m.RLock()
	defer m.RUnlock()

	select {
	case m.newBlocks <- &chainview.FilteredBlock{
		Hash:         hash,
		Height:       height,
		Transactions: txns,
	}:
	case <-m.quit:
		return
	}
	
	// Do not ack the block if our notify channel is nil. 
	if m.notifyBlockAck==nil{
		return
	}

	select{
		case m.notifyBlockAck <- struct{}{}:
		case time.After(defaultTimeout):
			t.Fatal("expected block delivered")
	}
}
```

Then in the test, we just wait to receive on `m.notifyBlockAck` rather than sleeping. More code, but it'll make for less flakey tests in the long run!

Reply:
good call, thanks for the explanation @carlaKC ! Added that to notifystaleblock as well